cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,817 @@
+#    )
+#
+#  # Wrapper target for the merge command.
+#  add_custom_target(
+#    ${container}_hex
+#    ALL DEPENDS
+#    ${PROJECT_BINARY_DIR}/${container}.hex
+#    )
+
+endforeach()
+
+# Start - Code related to network core update. Multi image updates are part of NCSDK-17807
+#if (CONFIG_SECURE_BOOT AND CONFIG_BOOTLOADER_MCUBOOT)
+#  # Create symbols for the offsets required for moving test update hex files
+#  # to MCUBoots secondary slot. This is needed because objcopy does not
+#  # support arithmetic expressions as argument (e.g. '0x100+0x200'), and all
+#  # of the symbols used to generate the offset is only available as a
+#  # generator expression when MCUBoots cmake code exectues. This because
+#  # partition manager is performed as the last step in the configuration stage.
+#  math(EXPR s0_offset "${PM_MCUBOOT_SECONDARY_ADDRESS} - ${PM_S0_ADDRESS}")
+#  math(EXPR s1_offset "${PM_MCUBOOT_SECONDARY_ADDRESS} - ${PM_S1_ADDRESS}")
+#
+#  set_property(
+#    TARGET partition_manager
+#    PROPERTY s0_TO_SECONDARY
+#    ${s0_offset}
+#    )
+#  set_property(
+#    TARGET partition_manager
+#    PROPERTY s1_TO_SECONDARY
+#    ${s1_offset}
+#    )
+#endif()
+# End - Code related to network core update. Multi image updates are part of NCSDK-17807
+
+# Always add main partition file to list.
+list(APPEND pm_out_partition_file ${APPLICATION_BINARY_DIR}/partitions.yml)
+list(APPEND pm_out_region_file    ${APPLICATION_BINARY_DIR}/regions.yml)
+
+if (is_dynamic_partition_in_domain)
+  # Nothing is built as child.
+  # We have all required info available, just need to use them.
+  # We are being built as sub image.
+  # Expose the generated partition manager configuration files to parent image.
+  # This is used by the root image to create the global configuration in
+  # pm_config.h.
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOMAIN_PARTITIONS ${pm_out_partition_file})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOMAIN_REGIONS ${pm_out_region_file})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOMAIN_HEADER_FILES ${header_files})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOMAIN_IMAGES ${prefixed_images})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_HEX_FILE ${PROJECT_BINARY_DIR}/${merged}.hex)
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_DOTCONF_FILES ${pm_out_dotconf_file})
+  set_shared(IMAGE ${DOMAIN} PROPERTY PM_APP_HEX ${PROJECT_BINARY_DIR}/app.hex)
+  set_shared(IMAGE ${IMAGE_NAME} APPEND PROPERTY BUILD_BYPRODUCTS ${PROJECT_BINARY_DIR}/${merged}.hex)
+  if(CONFIG_SECURE_BOOT)
+    # Only when secure boot is enabled the app will be signed.
+    set_shared(IMAGE ${DOMAIN} PROPERTY PM_SIGNED_APP_HEX ${PROJECT_BINARY_DIR}/signed_by_b0_app.hex)
+  endif()
+else()
+  # This is the root image, generate the global pm_config.h
+  # First, include the shared properties for all child images.
+  if (PM_DOMAINS)
+    # We ensure the existence of PM_DOMAINS to support older cmake versions.
+    # When version >= 3.17 is required this check can be removed.
```

Comment:
Zephyr requires 3.20 so can we get rid of code targeting < 3.20?

Reply:
removed